scripts/build-core.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -e
+
+MESG_SERVICE_SERVER=ipfs.app.mesg.com
+
+LDFLAGS="-X 'github.com/mesg-foundation/core/version.Version=$version'"
+
+# upload system services
+for s in systemservices/* ; do
+  if [ -d "$s" ]; then
+    pushd $s > /dev/null
+    name=$(basename "$s")
+    varname="${name^}"
+    archive="$name.tar.gz"
+
+    id=$(
+      tar -czf - --exclude-from=.dockerignore . |
+      curl -s -F "file=@-;filename=${archive}" http://$MESG_SERVICE_SERVER:5001/api/v0/add |
+      jq -r .Hash
+    )
+    LDFLAGS+=" -X 'github.com/mesg-foundation/core/config.${varname}URL=http://$MESG_SERVICE_SERVER:8080/ipfs/$id'"
```

Comment:
```suggestion
    LDFLAGS+=" -X 'github.com/mesg-foundation/core/config.${varname}URL=https://gateway.ipfs.io/ipfs/$id'"
```

can you change to use ipfs gateway for download? so like this, we use ipfs gateway as a cache

Reply:
I can but it might take a long time for ipfs to find our hash (I've tested on my local machine and it took like half an hour)

Reply:
It's true that there is a delay (like 5min in my case) before the file can be retrieve using ipfs gateway...
@antho1404 what do you think?

Reply:
Why is this so long? Could we ping ipfs gateway in that script to make sure that they start to fetch is as soon as possible? Will this be helpful? What's the reason for this delay?

Reply:
From https://docs.ipfs.io/introduction/usage/

> Note: depending on the state of the network, curl may take a while. The public gateways may be overloaded or having a hard time reaching you.

So probably node discovery + node sync + hash searching may take a while, and if you try to get the hash it's like ping but you have no guarantee when public node find the hash

Reply:
let's keep it like that for now if there is a delay with ipfs gateway and no delay with our node let's use our node